contracts/KeyManager.sol

```diff
@@ -93,7 +93,7 @@ contract KeyManager {
     // check if key exists
     require(_keys[key].purposes.length > 0, "Key does not exit");
 
-    _keys[key].revokedAt = block.number;
+    _keys[key].revokedAt = uint32(block.number);
```

Comment:
When do we add the second check?

Only allow revoking the key if the key is not already revoked?